test/gt2py_fo_exec.jl

```diff
@@ -19,73 +20,147 @@ macro to_py(expr::Expr)
     return res
 end
 
-# Setup ------------------------------------------------------------------------------------------
+# ========================================
+# ============== Utility =================
+# ========================================
 
-edge_to_cell_table = [
-    [1  -1];
-    [3  -1];
-    [3  -1];
-    [4  -1];
-    [5  -1];
-    [6  -1];
-    [1  6];
-    [1  2];
-    [2  3];
-    [2  4];
-    [4  5];
-    [5  6]
-]
+struct ConnectivityData
+    edge_to_cell_table::Matrix{Integer}
+    cell_to_edge_table::Matrix{Integer}
+    E2C_offset_provider::Connectivity
+    C2E_offset_provider::Connectivity
+    offset_provider::Dict{String,Connectivity}
+end
 
-cell_to_edge_table = [
-    [1   7   8];
-    [8   9  10];
-    [2   3   9];
-    [4  10  11];
-    [5  11  12];
-    [6   7  12]
-]
+"""
+    testwrapper(setupfunc::Union{Function,Nothing}, testfunc::Function, args...)
 
+Wrapper function to facilitate testing with optional setup.
 
-E2C_offset_provider = Connectivity(edge_to_cell_table, Cell, Edge, 2)
-C2E_offset_provider = Connectivity(cell_to_edge_table, Edge, Cell, 3)
+# Arguments
+- `setupfunc::Union{Function,Nothing}`: An optional setup function. If provided, it will be called before the test function. 
+                                        If `nothing`, the test function is called directly.
+- `testfunc::Function`: The test function to be executed.
+- `args...`: Additional arguments to be passed to the test function.
 
-offset_provider = Dict{String, Connectivity}(
-                   "E2C" => E2C_offset_provider,
-                   "C2E" => C2E_offset_provider
-                )
+# Usage
+- If `setupfunc` is provided, it should return the necessary data that `testfunc` will use. 
+  The returned data will be passed as the first argument to `testfunc`, followed by `args...`.
+- If `setupfunc` is `nothing`, `testfunc` will be called directly with `args...`.
 
+# Examples
 
-# Tests ------------------------------------------------
-function test_gt4py_fo_exec()
-    a = Field(Cell, collect(1.:15.))
-    b = Field(Cell, collect(-1.:-1:-15.))
+## Example 1: Without Setup Function
+```julia
+function mytest(args...)
+    println("Running test with arguments: ", args)
+end
+
+testwrapper(nothing, mytest, 1, 2, 3)
+# Output: Running test with arguments: (1, 2, 3)
+```
+
+## Example 2: With Setup Function
+```julia
+function setup()
+    return "setup data"
+end
+
+function mytest(data, args...)
+    println("Setup data: ", data)
+    println("Running test with arguments: ", args)
+end
+
+testwrapper(setup, mytest, 1, 2, 3)
```

Comment:
If you put the wrapper inside of `src`, add it to the documentation (e.g. a new section testing) you can run the doctests automatically (https://documenter.juliadocs.org/stable/man/doctests/).